tests/components/motionblinds_ble/__init__.py

```diff
@@ -1 +1,61 @@
 """Tests for the Motionblinds Bluetooth integration."""
+
+from unittest.mock import patch
+
+from motionblindsble.const import MotionBlindType
+
+from homeassistant.components.motionblinds_ble import async_setup_entry
+from homeassistant.components.motionblinds_ble.const import (
+    CONF_BLIND_TYPE,
+    CONF_LOCAL_NAME,
+    CONF_MAC_CODE,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.service_info.bluetooth import BluetoothServiceInfo
+
+from tests.common import MockConfigEntry
+
+FIXTURE_SERVICE_INFO = BluetoothServiceInfo(
+    name="MOTION_CCCC",
+    address="cc:cc:cc:cc:cc:cc",
+    rssi=-63,
+    service_data={},
+    manufacturer_data={},
+    service_uuids=[],
+    source="local",
+)
+
+
+async def setup_platform(
+    hass: HomeAssistant,
+    platforms: list[Platform],
+    blind_type: MotionBlindType = MotionBlindType.ROLLER,
+) -> tuple[MockConfigEntry, str]:
+    """Mock a fully setup config entry."""
+
+    config_entry = MockConfigEntry(
+        title="mock_title",
+        domain=DOMAIN,
+        unique_id="cc:cc:cc:cc:cc:cc",
+        data={
+            CONF_ADDRESS: "cc:cc:cc:cc:cc:cc",
+            CONF_LOCAL_NAME: "Motionblind CCCC",
+            CONF_MAC_CODE: "CCCC",
+            CONF_BLIND_TYPE: blind_type.name.lower(),
+        },
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.motionblinds_ble.PLATFORMS", platforms):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await async_setup_entry(hass, config_entry)
```

Comment:
This should not be called manually

Reply:
How else can the `async_setup_entry` be called, considering the integration includes both an `async_setup` and `async_setup_entry`?

Reply:
the line above makes HA setup the integration. It will first call `async_setup` and then it will call `async_setup_entry`

Reply:
> the line above makes HA setup the integration. It will first call `async_setup` and then it will call `async_setup_entry`

Thing is I could not get the tests to work without the `await async_setup_entry(hass, config_entry)` line. Only using `await hass.config_entries.async_setup(config_entry.entry_id)` does not trigger `async_setup_entry` and therefore the platforms are not setup

Reply:
That sounds like an issue on itself tbh

Reply:
Why would it be? Were async_setup and async_setup_entry not designed to work together? If so, why would you not be able to have a general integration setup and and a setup for each entry?

Reply:
Well that's the thing, it should work together. when an entry is set up, `async_setup` is run, and then `async_setup_entry`. That's a given. If that is not the case, something is going wrong. Maybe something is not patched.

Reply:
Hmm not sure why. Without the `await async_setup_entry(hass, config_entry)` pretty much all tests fail as the platforms are not setup. I did check debug logs and the async_setup_entry is not called at all.